### ISSUE TITLE
[BUG] Spamassassin no longer working due to spamhaus policy change

### ISSUE THREAD

**Issue by tve** [docker-mailserver/docker-mailserver#2918]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

spamassassin queries to spamhaus

### What happened and when does this occur?

```Markdown
Queries to the spamhaus blocking lists are intermittently failing. The result is log entries of the form:

Dec  1 08:22:29 mail postfix/smtpd[224054]: NOQUEUE: reject: RCPT from o8.email.nextdoor.com[167.89.48.102]: 554 5.7.1 Service unavailable; Client host [167.89.48.102] blocked using zen.spamhaus.org; Error: open resolver; https://www.spamhaus.org/returnc/pub/35.80.40.194; from=<bounces+1740987-5fc4-xxx=voneicken.com@rs.email.nextdoor.com> to=<xxx@voneicken.com> proto=ESMTP helo=<o8.email.nextdoor.com>

Dec  1 08:29:53 mail postfix/smtpd[225785]: NOQUEUE: reject: RCPT from s4.csa2.acemsa1.com[52.128.40.9]: 554 5.7.1 Service unavailable; Client host [52.128.40.9] blocked using zen.spamhaus.org; from=<bounce-2252638-609-18526-xxx=voneicken.com@s4.csa2.acemsa1.com> to=<xxx@voneicken.com> proto=ESMTP helo=<s4.csa2.acemsa1.com>
```

### What did you expect to happen?

```Markdown
I expected the queries to spamhaus to work.
```

### How do we replicate the issue?

```Markdown
My DMS container is running in AWS and uses the AWS DNS resolvers. Clicking through the link in the the first error message above eventually gets to https://www.spamhaus.com/resource-center/successfully-accessing-spamhauss-free-block-lists-using-a-public-dns/
From that page I gather that my DMS is being blocked because (a) it uses a public DNS resolver and (b) it runs in a cloud hosting environment.
I assume that the issue can be replicated by setting up DMS with 8.8.8.8 as DNS resolver.
```

### DMS version

V11.2.0

### What operating system is DMS running on?

Linux

### Which operating system version?

Ubuntu 22.04

### What instruction set architecture is DMS running on?

x86_64 / AMD64

### What container orchestration tool are you using?

Docker

### docker-compose.yml

```yml
version: '2'

services:
  mail:
    image: docker.io/mailserver/docker-mailserver:latest
    # build: .
    hostname: mail
    domainname: voneicken.com
    container_name: mail
    cap_add:
      - NET_ADMIN
    restart: always
    stop_grace_period: 1m
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
    ports:
    - "25:25" # SMTP w/STARTTLS
    - "465:465" # ESMTP w/TLS
    - "587:587" # ESMTP w/STARTTLS
    - "172.22.154.1:993:993" # IMAP w/TLS
    - "172.22.154.1:4190:4190" # SIEVE
    #- "143:143" # IMAP w/out TLS or STARTTLS
    volumes:
    - /home/mail:/var/mail
    - /home/mail-state:/var/mail-state
    - /home/mail-state/letsencrypt/etc:/etc/letsencrypt
    - ./config/:/tmp/docker-mailserver/
    - /var/log/mail:/var/log/mail
    - /etc/localtime:/etc/localtime:ro
    environment:
    - ENABLE_CLAMAV=1
    - ENABLE_SPAMASSASSIN=1
    - ENABLE_FAIL2BAN=1
    - ENABLE_MANAGESIEVE=1
    - ENABLE_SASLAUTHD=0
    - ENABLE_QUOTAS=0
    - POSTFIX_MESSAGE_SIZE_LIMIT=40000000
    - SSL_TYPE=letsencrypt
    - RELAY_HOST=email-smtp.us-east-1.amazonaws.com
    - RELAY_PORT=587
    - RELAY_USER=...
    - RELAY_PASSWORD=...
    - ONE_DIR=1
    - REPORT_RECIPIENT=xxx@voneicken.com
    - REPORT_INTERVAL=weekly
```

### Other relevant information

```Markdown
The solution offered by spamhaus is to sign up for a free account, which I'm doing, but using that account requires a plugin to spamassassin and presumably some configuration of that plugin.
It seems to me that I must not be the only user of DMS that is experiencing the issue, however, so far I've noticed that it is intermittent, i.e., spamhaus is not blocking all the queries, only some fraction, so emails continue to come through, they're just often delayed by minutes until the sending MTA retries.
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [X] I am rather experienced with docker
- [X] I am inexperienced with mail servers
- [ ] I am rather experienced with mail servers
- [ ] I am uncomfortable with the CLI
- [X] I am rather comfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by tve** [comment]
I had started this issue as discussion: #2917

**Comment by tve** [comment]
I thought it was fixed, see discussion thread, but it's not. Sometimes spamhaus responds and sometimes it doesn't. I'll have to set `ENABLE_DNSBL=0` and hope that prevents the errors.
```
Dec  1 21:46:52 mail postfix/smtpd[13477]: NOQUEUE: reject: RCPT from b224-8.smtp-
out.eu-central-1.amazonses.com[69.169.224.8]: 554 5.7.1 Service unavailable; Clien
t host [69.169.224.8] blocked using zen.spamhaus.org; from=<01070184d15f87fe-3c219
93b-cadd-4435-8d2b-0849ded09378-000000@notify.surfski.info> to=<xxx@voneicken.com>
 proto=ESMTP helo=<b224-8.smtp-out.eu-central-1.amazonses.com>                    
```

I'm not clear on whether postfix and postscreebn both contact spamhaus and whether they're performing the same function (i.e. are redundant). I don't see postscreen being affected by the issue, or at least it's not spewing error messages. E.g.:
```
Dec  1 21:47:52 mail postfix/postscreen[13469]: CONNECT from [114.216.202.213]:257
03 to [172.26.0.2]:25                                                             
Dec  1 21:47:53 mail postfix/dnsblog[13473]: addr 114.216.202.213 listed by domain
 zen.spamhaus.org as 127.255.255.254                                              
Dec  1 21:47:53 mail postfix/dnsblog[13476]: addr 114.216.202.213 listed by domain
 list.dnswl.org as 127.0.10.3                                                     
Dec  1 21:47:53 mail postfix/postscreen[13469]: PREGREET 11 after 0.22 from [114.2
16.202.213]:25703: EHLO User\r\n                                                  
```

**Comment by georglauterbach** [comment]
I see where the issue here really is:

As @tve has reported, Postscreen is not showing errors because I fixed this in #2890. Results from Spamhaus are only accounted for if the return value if `127.0.0.[2..11]`. For SpamAssassin though, there is no such setting in DMS..

This is - more or less - not an issue with DMS itself, but due to the fact you're using an public resolver and hence, an issue with DNSBLs at large.

**The proper fix is this**: run a recursive resolver yourself. I faced this problem too, but resolved it but running a recursive resolver myself. Because I figured others might want to do the same, I published it on GitHub: (shameless self-promotion incoming) [`georglauterbach/k8s-dns`](https://github.com/georglauterbach/k8s-dns/). This is a container image that runs Bind9, and the configuration in the repository enables recursive queries. You can use a similar configuration. The container is straight-forward to use, just mount your configuration properly. It was designed with Kubernetes in mind, hence the ports are 8053, but you can probably remap the ports with Docker Compose too to port 53 (TCP & UDP).
